### DISCUSSION TITLE
Unique constraints and one-to-one relationships are not possible in Budibase?

### DISCUSSION THREAD

**Discussion by dnk8n** [Budibase/budibase#10076] [Help] [answered]
In the documentation there is mention of one-to-one relationships - https://docs.budibase.com/docs/relationships#one-to-one but in the video demo, and in the options, only one-to-many is possible.

Also, I have not found a way to enforce a unique constraint at the database level.

For a properly modeled application, is it better to define schema and views independently of Budibase? I need standard best practices and the full power of Postgres to be honored, as I am only using Budibase for data entry. I need to be able to run advanced queries on the resulting database.

**Comment by melohagan** [comment, marked as answer]
Hey @dnk8n 

You're right, only One->Many relationships are possible, however you can use this as if it was a One->One, however you would need to manage that in your app.

Defining your schema in Postgres will mean that any constraints defined be respected by your Budibase app, as Budibase simply passes queries to your Postgres database in the backend. 

Please note that Postgres view are currently not supported: https://github.com/Budibase/budibase/issues/6216

If you need to run advanced Postgres queries, then you can do that through [Custom queries](https://docs.budibase.com/docs/data-sources#custom-queries)
